lib/util/host-rules.ts

```diff
@@ -7,27 +7,46 @@ import { parseUrl, validateUrl } from './url';
 
 let hostRules: HostRule[] = [];
 
-const legacyHostFields = ['hostName', 'domainName', 'baseUrl'];
+interface LegacyHostRule {
+  hostName?: string;
+  domainName?: string;
+  baseUrl?: string;
+}
 
-export function add(params: HostRule): void {
-  const rule = clone(params);
-  const matchedFields = legacyHostFields.filter((field) => rule[field]);
-  if (matchedFields.length) {
-    if (rule.matchHost || matchedFields.length > 1) {
-      throw new Error(
-        `hostRules cannot contain more than one host-matching field - use "matchHost" only.`
-      );
+function isHostRuleField(input: string): input is keyof HostRule {
+  return !['hostName', 'domainName', 'baseUrl'].includes(input);
```

Comment:
Please move this array back to module scope, so we don't create a new on every function call